### ISSUE TITLE
pikaur suddenly fails to build AUR packages, complains about executing makepkg as root

### ISSUE THREAD

**Issue by Baerbeisser** [actionless/pikaur#710]
```
Pikaur v1.14.7
Pacman v6.0.2 - libalpm v13.0.2 - pyalpm v0.10.6

```

##### Prerequisites:
 [build]
alwaysusedynamicusers = no

##### Description:
Using Artix-Dinit (no Systemd), pikaur fails to build AUR packages, with error `Running makepkg as root is not allowed as it can cause permanent, catastrophic damage to your system.`. Updates from the repo install fine.
This started appearing in the last few days, did build fine before.
Running `python3 pikaur.py -S pikaur` from git clone doesn't change anything.
Might this be related to Systemd dynamic users, did something change in pikaur?

##### Attached log:

pikaur is just an example here, fails with all other packages too.

```
:: debug: main: Setting stdout to utf-8...
:: debug: main: already set - nothing to do
:: debug: main: Setting stderr to utf-8...
:: debug: main: already set - nothing to do
:: debug: main: Pikaur operation found for sys.argv=['/sbin/pikaur', '-S', 'pikaur', '--verbose', '--pikaur-debug']: cli_install_packages
(sudo) Password: 
:: debug: install_info_fetcher: Gonna fetch install info for:
    install_package_names=['pikaur']
    not_found_repo_pkgs_names=[]
    pkgbuilds_packagelists={}
    manually_excluded_packages_names=[]

:: debug: install_info_fetcher: Gonna get repo pkgs install info...
Reading repository package databases...
Reading local package database...
:: debug: install_info_fetcher: Checking if 'pkg_lines=['pikaur']' is installable:
=> pacman --color=always --sync pikaur --print-format %r/%n
:: debug: install_info_fetcher: Check failed - gonna check it separately:
:: debug: install_info_fetcher: Checking if 'pikaur' exists in the repo:
=> pacman --color=always --sync pikaur --print-format %r/%n --nodeps --nodeps
:: debug: install_info_fetcher:   'pikaur' is NOT found in the repo.
:: debug: install_info_fetcher: Check partially passed - gonna get install infos:
:: debug: install_info_fetcher: gonna get AUR pkgs install info for:
    aur_packages_versionmatchers=['pikaur']
    self.aur_updates_install_info=[]
    aur_packages_names_to_versions={'pikaur': <VersionMatcher pikaur['cmp_default']None>}
=> GET https://aur.archlinux.org/rpc/?v=5&type=info&arg[]=pikaur
:: debug: install_info_fetcher: found AUR pkgs:
    aur_pkg_list=[<AURPackageInfo "pikaur" 1.14.7-1>]
    not_found_aur_pkgs=[]

:: debug: install_info_fetcher: got AUR pkgs install info: self.aur_updates_install_info=[<AURInstallInfo "pikaur" 1.14.7-1 -> 1.14.7-1>]
Resolving AUR dependencies...
=> pacman --color=always --deptest pyalpm git python-wheel python-build python-installer python-setuptools python-markdown-it-py
=> pacman --color=always --sync --print-format=%% --nodeps python-installer python-build
=> pacman --color=always --sync python-build --print-format %r/%n --nodeps --nodeps
=> pacman --color=always --sync python-installer --print-format %r/%n --nodeps --nodeps
:: debug: install_info_fetcher: Checking if 'pkg_lines=['python-installer', 'python-build']' is installable:
=> pacman --color=always --sync --needed python-installer python-build --print-format %r/%n
:: debug: install_info_fetcher: Check passed - gonna get install infos:
:: debug: install_cli: self.install_info.all_install_info_containers=([], [], [], [], [<RepoInstallInfo "python-installer"  -> 0.7.0-1>, <RepoInstallInfo "python-pyproject-hooks"  -> 1.0.0-1>, <RepoInstallInfo "python-build"  -> 0.10.0-1>], [], [<AURInstallInfo "pikaur" 1.14.7-1 -> 1.14.7-1>], [])

:: New dependencies will be installed from third-party repository:
 python-build (for pikaur)                                  -> 0.10.0-1
 python-installer (for pikaur)                              -> 0.7.0-1
 python-pyproject-hooks (for python-build)                  -> 1.0.0-1

:: AUR package will be installed:
 pikaur                                1.14.7-1             -> 1.14.7-1

:: debug: prompt: Gonna get input from user...
:: debug: prompt_nolock: Restoring TTY...
:: debug: prompt_nolock: Using standard input reader...
:: Proceed with installation? [Y/n] 
:: [v]iew package details   [m]anually select packages
>> 
:: debug: prompt_nolock: Reverting to prev TTY state...
:: debug: prompt: No answer provided - using "Y".
=> git -C /media/Games/Profile/.local/share/pikaur/aur_repos/pikaur pull origin master
=> pacman --color=always --deptest pyalpm git
=> pacman --color=always --deptest python-wheel python-build python-installer python-setuptools python-markdown-it-py
looking for conflicting AUR packages...
:: warning: Not showing diff for pikaur package (installing for the first time)
:: debug: prompt: Gonna get input from user...
:: debug: prompt_nolock: Restoring TTY...
:: debug: prompt_nolock: Using standard input reader...
Do you want to edit PKGBUILD for pikaur package? [y/N] 
:: debug: prompt_nolock: Reverting to prev TTY state...
:: debug: prompt: No answer provided - using "N".

:: debug: install_cli: Gonna build self.package_builds_by_name={'pikaur': <PackageBuild "pikaur" ['pikaur']>}
:: debug: install_cli: Gonna build pkg_build.package_names=['pikaur']
=> mkdir -p /media/Games/Profile/.cache/pikaur/build/pikaur
=> cp -r /media/Games/Profile/.local/share/pikaur/aur_repos/pikaur/CHANGELOG /media/Games/Profile/.local/share/pikaur/aur_repos/pikaur/PKGBUILD /media/Games/Profile/.local/share/pikaur/aur_repos/pikaur/.SRCINFO /media/Games/Profile/.cache/pikaur/build/pikaur/
:: debug: pacman: Discarding local cache...
Reading local package database...
=> pacman --color=always --deptest pyalpm git
=> pacman --color=always --deptest python-wheel python-build python-installer python-setuptools python-markdown-it-py
:: Installing repository dependencies for pikaur:
pikspect => sudo pacman --color=always --sync --asdeps python-build python-installer
:: debug: pikspect: Opening virtual terminal...
resolving dependencies...
looking for conflicting packages...

Package (3)                    New Version  Net Change

galaxy/python-pyproject-hooks  1.0.0-1        0.07 MiB
galaxy/python-build            0.10.0-1       1.08 MiB
galaxy/python-installer        0.7.0-1        0.77 MiB

Total Installed Size:  1.92 MiB

:: Proceed with installation? [Y/n] Y
(3/3) checking keys in keyring                                                                   [--------------------------------------------------------] 100%
(3/3) checking package integrity                                                                 [--------------------------------------------------------] 100%
(3/3) loading package files                                                                      [--------------------------------------------------------] 100%
(3/3) checking for file conflicts                                                                [--------------------------------------------------------] 100%
(3/3) checking available disk space                                                              [--------------------------------------------------------] 100%
:: Processing package changes...
(1/3) installing python-pyproject-hooks                                                          [--------------------------------------------------------] 100%
(2/3) installing python-build                                                                    [--------------------------------------------------------] 100%
Optional dependencies for python-build
    python-virtualenv: Use virtualenv for build isolation
(3/3) installing python-installer                                                                [--------------------------------------------------------] 100%
:: debug: pacman: Discarding local cache...
Reading local package database...
=> makepkg --packagelist

:: Starting the build:
=> makepkg --force
==> ERROR: Running makepkg as root is not allowed as it can cause permanent,
catastrophic damage to your system.

Command 'makepkg --force' failed to execute.
:: debug: build: Gonna compute diff of installed pkgs
:: debug: build: deps_packages_installed={'python-build', 'python-pyproject-hooks', 'python-installer'}
:: debug: build: deps_packages_removed=set()
:: Removing already installed dependencies for pikaur:
pikspect => sudo pacman --color=always --remove python-build python-pyproject-hooks python-installer
:: debug: pikspect: Opening virtual terminal...
checking dependencies...

Package (3)             Old Version  Net Change

python-build            0.10.0-1      -1.08 MiB
python-installer        0.7.0-1       -0.77 MiB
python-pyproject-hooks  1.0.0-1       -0.07 MiB

Total Removed Size:  1.92 MiB

:: Do you want to remove these packages? [Y/n] Y
:: Processing package changes...
(1/3) removing python-installer                                                                  [--------------------------------------------------------] 100%
(2/3) removing python-build                                                                      [--------------------------------------------------------] 100%
(3/3) removing python-pyproject-hooks                                                            [--------------------------------------------------------] 100%
:: debug: pacman: Discarding local cache...

Can't build 'pikaur'.

:: debug: install_cli: discarding install info for pkg... pikaur
:: debug: install_cli: ignoring pkg... pikaur
:: debug: install_info_fetcher: discarding canceled_pkg_name='pikaur'
:: debug: install_cli: discarded install info for pkg: pikaur
Failed to build following packages:
pikaur
:: debug: main: Restoring stdout...
:: debug: main: nothing to do
:: debug: main: Restoring stderr...
:: debug: main: nothing to do


```

**Comment by actionless** [comment]
fixed in 99fa4549e9f25661c65c7c031027033c9be3dadc

**Comment by Baerbeisser** [comment]
It happens again.
